[Perf Framework] BatchPerfTest & Service Bus Receive Batch perf test (#18811)
Azure/azure-sdk-for-js: sdk/test-utils/perf/CHANGELOG.md

## 1.0.0 (Unreleased)

### 2021-11-24

- Adds `PerfTestBase` and `BatchPerfTest`.

>   - `PerfTestBase` abstract class with the skeleton of the perf tests.

Reviewer: ```suggestion
  - `PerfTestBase` is an abstract class with the skeleton needed to build other classes representing performance tests.
```

Something like this, perhaps?

>   - `BatchPerfTest` extends `PerfTestBase`, `BatchPerfTest` enables writing perf tests with more flexibility where the number of operations are dynamic for the method/call being tested.

Reviewer: ```suggestion
  - `BatchPerfTest` (which extends `PerfTestBase`) enables writing performance tests where the number of operations are dynamic.
```

>   - `PerfTest` now extends `BatchPerfTest`, a single call in the "run" method for the method being tested counts as one operation - has no breaking changes now and is supposed to work as it did before.

Reviewer: nit

```suggestion
  - `PerfTest` now extends `BatchPerfTest`. Going forward, a single call in the "run" method for the method being tested counts as one operation. This is not a breaking change, so existing performance tests will continue to work without changes.
```

Author: > Going forward, a single call in the "run" method for the method being tested counts as one operation.

Not just going forward, it has always been the same.

Updating to 
> - `PerfTest` now extends `BatchPerfTest`. A single call in the "run" method for the method being tested counts as one operation. `PerfTest` related updates are not breaking, so existing performance tests will continue to work without changes.

Author: ```suggestion
  - `PerfTest` now extends `BatchPerfTest`. A single call in the "run" method for the method being tested counts as one operation. `PerfTest` related updates are not breaking, so existing performance tests will continue to work without changes.
```

> - New template type `ParsedPerfOptions<TOptions = Record<string, unknown>>` for the parsed options, which doesn't require to add/bypass the "value" existence check.

Reviewer: ```suggestion
- A new template type `ParsedPerfOptions<TOptions = Record<string, unknown>>` was added to the parsed options. This type doesn't require to add or bypass the `value` check.
```

Author: ```suggestion
- A new template type `ParsedPerfOptions<TOptions = Record<string, unknown>>` is added to the parsed options. This type doesn't require adding or bypassing the `value` check.
```
